### PR TITLE
add OGC API - EDR client

### PR DESCRIPTION
**Pull request by tomkralidis**
FYI this PR should be merged once the pygeoapi demo server is updated via [this PR](https://github.com/geopython/pygeoapi/pull/1431).

### CODE REVIEW

File: owslib/ogcapi/edr.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+from typing import BinaryIO
+
+from owslib.ogcapi.features import Features
+from owslib.util import Authentication
+
+LOGGER = logging.getLogger(__name__)
+
+
+class EnvironmentalDataRetrieval(Features):
+    """Abstraction for OGC API - Envirionmental Data Retrieval"""
```
Comment (huard) [inline owslib/ogcapi/edr.py:20]:
```suggestion
    """Abstraction for OGC API - Environmental Data Retrieval"""
```

File: owslib/ogcapi/edr.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    def __init__(self, url: str, json_: str = None, timeout: int = 30,
+                 headers: dict = None, auth: Authentication = None):
+        __doc__ = Features.__doc__  # noqa
```
Comment (huard) [inline owslib/ogcapi/edr.py:24]: Isn't there a need for a custom docstring ?

File: owslib/ogcapi/edr.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        super().__init__(url, json_, timeout, headers, auth)
+
+    def data(self) -> list:
+        """
+        implements /collections filtered on EDR data resources
```
Comment (huard) [inline owslib/ogcapi/edr.py:29]:
```suggestion
        Collections filtered on EDR data resources.
```

File: owslib/ogcapi/edr.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                if 'data' in l_['rel']:
+                    datas.append(c_['id'])
+                    break
+
+        return datas
+
+    def query_data(self, collection_id: str,
+                   query_type: str, **kwargs: dict) -> BinaryIO:
+        """
+        implements /collection/{collectionId}/coverage/
```
Comment (huard) [inline owslib/ogcapi/edr.py:48]: From someone who's not familiar with the API, I suggest the docstring should start with what the function does in simple terms.

**Review by huard — APPROVED** [review]

File: owslib/ogcapi/edr.py (hunk 2 above)
Comment (tomkralidis) [reply owslib/ogcapi/edr.py:24]: Not at this level, given the building blocks of OGC APIs (collections are the same across APIS). docstrings are more specific deeper into the endpoints.

**Pull request merged**